test/integration/change-streams/change_stream.test.ts

```diff
@@ -1563,3 +1493,584 @@ describe('Change Streams', function () {
+
+          expect(aggregateEvents).to.have.lengthOf(2);
+        }
+      );
+    }
+
+    it(
+      'maintains the change stream options on resume',
+      { requires: { topology: '!single', mongodb: '>=4.2' } },
+      async function () {
+        changeStream = collection.watch([], changeStreamResumeOptions);
+
+        await client.db('admin').command(<FailPoint>{
+          configureFailPoint: is4_2Server(this.configuration.version)
+            ? 'failCommand'
+            : 'failGetMoreAfterCursorCheckout',
+          mode: { times: 1 },
+          data: {
+            failCommands: ['getMore'],
+            errorCode: resumableErrorCodes[0].code
+          }
+        });
+
+        expect(changeStream.cursor)
+          .to.have.property('options')
+          .that.containSubset(changeStreamResumeOptions);
+
+        const changes = once(changeStream, 'change');
+        await once(changeStream.cursor, 'init');
+
+        await collection.insertOne({ name: 'bailey' });
+
+        await changes;
+
+        expect(changeStream.cursor)
+          .to.have.property('options')
+          .that.containSubset(changeStreamResumeOptions);
+      }
+    );
+
+    context('when the error is not a resumable error', function () {
+      it(
+        'does not resume',
+        { requires: { topology: '!single', mongodb: '>=4.2' } },
+        async function () {
+          changeStream = collection.watch([]);
+
+          const unresumableErrorCode = 1000;
+          await client.db('admin').command(<FailPoint>{
+            configureFailPoint: is4_2Server(this.configuration.version)
+              ? 'failCommand'
+              : 'failGetMoreAfterCursorCheckout',
+            mode: { times: 1 },
+            data: {
+              failCommands: ['getMore'],
+              errorCode: unresumableErrorCode
+            }
+          });
+
+          const willBeError = once(changeStream, 'change').catch(error => error);
+          await once(changeStream.cursor, 'init');
+          await collection.insertOne({ name: 'bailey' });
+
+          const error = await willBeError;
+
+          expect(error).to.be.instanceOf(MongoServerError);
+          expect(aggregateEvents).to.have.lengthOf(1);
+        }
+      );
+    });
+  });
+
+  it(
+    'caches the server version after the initial aggregate call',
+    { requires: { topology: '!single' } },
+    async function () {
+      changeStream = collection.watch([], changeStreamResumeOptions);
+      await initIteratorMode(changeStream);
+
+      expect(changeStream.cursor.maxWireVersion).not.to.be.undefined;
+    }
+  );
+
+  it(
+    'caches the server version after each getMore call',
+    { requires: { topology: '!single' } },
+    async function () {
+      changeStream = collection.watch([], changeStreamResumeOptions);
+      await initIteratorMode(changeStream);
+
+      const maxWireVersion = changeStream.cursor.maxWireVersion;
+      changeStream.cursor.maxWireVersion = 20;
```

Comment:
are you trying to test that the version is overwritten after the update? if so, would it be better to set it to something that's super low instead? since in the future, the wire versions will go up? also, the test case name should maybe say that it `updates the cached server version`

I do wonder though why the test setup is different in this case vs the aggregate case; feels like maybe both of them should make more than one call in order to satisfy the "each" part of the test case description

Reply:
There's only ever one aggregate call (it just initializes the change stream), so the aggregate test should only initialize the cursor once.  As far as I know, initializing an initialized cursor is undefined behavior.

Good point about the `each` point - I'll update the test to call getMore again and confirm that the value has been set after each call.